ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Mempool/Query.hs

```diff
@@ -33,25 +32,32 @@ implGetSnapshotFor mpEnv slot ticked readUntickedTables = do
       -- We are looking for a snapshot exactly for the ledger state we already
       -- have cached, then just return it.
       pure . snapshotFromIS $ is
-    else do
-       let keys = Foldable.foldMap'
-                    getTransactionKeySets
-                    [ txForgetValidated . TxSeq.txTicketTx $ tx
-                    | tx <- TxSeq.toList $ isTxs is
-                    ]
-       values <- readUntickedTables keys
-       pure $ snapshotFromIS $
-         if pointHash (isTip is) == castHash (getTipHash ticked) && isSlotNo is == slot
-         then is
-         else newInternalState
-            $ revalidateTxsFor
-                capacityOverride
-                cfg
-                slot
-                ticked
-                values
-                (isLastTicketNo is)
-                (TxSeq.toList $ isTxs is)
+    else if pointHash (isTip is) == castHash (getTipHash ticked)
```

Comment:
I think there should be one call to `computeSnapshot`, where the `values` has been conditionally recovered either from `isTxValues` or from `isTxKeys`.